ext/drreg/drreg.c

```diff
@@ -420,7 +419,9 @@ drreg_event_bb_insert_late(void *drcontext, void *tag, instrlist_t *bb, instr_t
     /* Before each app read, or at end of bb, restore aflags to app value */
     uint aflags = (uint)(ptr_uint_t)drvector_get_entry(&pt->aflags.live, pt->live_idx);
     if (!pt->aflags.native &&
-        (drmgr_is_last_instr(drcontext, inst) ||
+        ((drmgr_is_last_instr(drcontext, inst) &&
+          !TEST(DRREG_IGNORE_BB_END_RESTORE, pt->bb_props)) ||
```

Comment:
Hmm, not restoring at bb end breaks rseq handling.  I think we would have to solve #3798 before this could be added.

Reply:
At the end of a basic block, the user needs to ensure that all registers are restored and have their original values. This is why DRREG_IGNORE_BB_END_RESTORE is coupled with drreg_restore_all(). It is the user's responsibility to call drreg_restore_all() when  using DRREG_IGNORE_BB_END_RESTORE. I hinted this in a comment in this PR.

Reply:
@derekbruening do you agree that ignoring bb end restoration won't hinder rseq handling? Essentially the user needs to ensure that  drreg_restore_all is called.

Reply:
If the user restores, yes.